src/python/doc/wasserstein_distance_user.rst

```diff
@@ -58,16 +58,29 @@ An index of -1 represents the diagonal.
     import gudhi.wasserstein
     import numpy as np
 
-    diag1 = np.array([[2.7, 3.7],[9.6, 14.],[34.2, 34.974]])
-    diag2 = np.array([[2.8, 4.45], [5, 6], [9.5, 14.1]])
-    cost, matching = gudhi.wasserstein.wasserstein_distance(diag1, diag2, matching=True, order=1., internal_p=2.)
-
-    message = "Wasserstein distance value = %.2f, optimal matching: %s"  %(cost, matching)
-    print(message)
+    dgm1 = np.array([[2.7, 3.7],[9.6, 14.],[34.2, 34.974]])
+    dgm2 = np.array([[2.8, 4.45], [5, 6], [9.5, 14.1]])
+    cost, matchings = gudhi.wasserstein.wasserstein_distance(diag1, diag2, matching=True, order=1., internal_p=2.)
+
+    message_cost = "Wasserstein distance value = %.2f" %cost
+    print(message_cost)
+    dgm1_to_diagonal = matchings[np.where(matchings[:,0] == -1)][:,1]
+    dgm2_to_diagonal = matchings[np.where(matchings[:,1] == -1)][:,0]
+    off_diagonal_match = np.delete(matchings, np.where(matchings == -1)[0], axis=0)
```

Comment:
An alternative would be `a[(a!=-1).all(axis=1)]`, but it isn't obviously better or worse.

Reply:
Ok to keep your version with delete here, but if you write an example or tutorial elsewhere, please use the variant `a[(a!=-1).all(axis=1)]` so users can see that one as well. Deal? (If you don't write one of those, then you have nothing to do)

Reply:
Deal. I plan to write a tutorial about barycenters once #167 got merged, it could be such an opportunity.